config/awesome/floppy/module/dynamic-wallpaper.lua

```diff
@@ -56,38 +67,213 @@ end
 
 -- Countdown variable
 -- In seconds
-the_countdown = nil
+local the_countdown = nil
+
+-- Parse seconds to HH:MM:SS
+local function parse_to_time(seconds)
+	-- DST ruined me :(
+	--return os.date("%H:%M:%S", seconds)
 
--- We will use a table for hour change and wallpaper string
--- Element 0 errm 1 will store the incoming/next scheduled time
--- Geez why the f is lua's array starts with `1`? lol
--- Element 2 will have the wallpaper file name
-local wall_data = {}
--- > Why array, you say? 
--- Because why not? I'm new to lua and I'm experimenting with it
+	local function format(str)
+		while #str < 2 do
+			str = '0' .. str
+		end
+
+		return str
+	end
+
+	local function convert(num)
+		return format(tostring(num))
+	end
+
+	local hours = convert(math.floor(seconds / 3600))
+	seconds = seconds - (hours * 3600)
+
+	local minutes = convert(math.floor(seconds / 60))
+	seconds = seconds - (minutes * 60)
+
+	local seconds = convert(math.floor(seconds))
+
+	return (hours .. ':' .. minutes .. ':' .. seconds)
+
+end
 
 -- Parse HH:MM:SS to seconds
 local parse_to_seconds = function(time)
 
   	-- Convert HH in HH:MM:SS
-  	hour_sec = tonumber(string.sub(time, 1, 2)) * 3600
+  	local hour_sec = tonumber(string.sub(time, 1, 2)) * 3600
 
   	-- Convert MM in HH:MM:SS
-  	min_sec = tonumber(string.sub(time, 4, 5)) * 60
+  	local min_sec = tonumber(string.sub(time, 4, 5)) * 60
 
 	-- Get SS in HH:MM:SS
-	get_sec = tonumber(string.sub(time, 7, 8))
+	local get_sec = tonumber(string.sub(time, 7, 8))
 
 	-- Return computed seconds
     return (hour_sec + min_sec + get_sec)
 end
 
 -- Get time difference
-local time_diff = function(current, schedule)
-	local diff = parse_to_seconds(current) - parse_to_seconds(schedule)
+local time_diff = function(future, past)
+	local diff = parse_to_seconds(future) - parse_to_seconds(past)
+	if diff < 0 then
+		diff = diff + (24 * 3600) --If time difference is negative, the future is meant for tomorrow
+	end
 	return diff
 end
 
+-- Split string based on pattern
+local function split(str, pat)
+	local t = {}
+	local i = 0
+	local j = string.find(str, pat)
+	while j ~= nil do
+		table.insert(t, string.sub(str, i + 1, j - 1))
+
+		i = j
+		j = string.find(str, pat, i + 1)
+	end
+	table.insert(t, string.sub(str, i + 1, #str))
+
+	return t
+end
```

Comment:
can you use `gears.string.split()` instead of rolling a custom function here?

https://awesomewm.org/apidoc/utility_libraries/gears.string.html#split

Reply:
As far as i can tell, `gears.string.split()` uses a single character instead of a lua pattern to delimit, which is probably why I made a custom `split()`. However, I've just noticed I can do what i was trying to do in the first place using `string.find()` and don't need the custom function anymore so I'll remove it